src/interfaces/records-read.ts

```diff
@@ -57,6 +75,28 @@ export class RecordsRead extends Message<RecordsReadMessage> {
     return new RecordsRead(message);
   }
 
+  /**
+   * Creates a Records interface Filter using either the `recordId` or `protocol` & `protocolPath` of the incoming Message.
+   * @param descriptor message descriptor with optional properties `recordId`, `protocol` and `protocolPath`
+   *
+   * @returns {Filter} with a Records interface as well as the appropriate filter params
+   * @throws {DwnError} when either `recordId` is missing, or both `protocol` and `protocolPath` are missing.
+   */
+  public static createFilter(descriptor: RecordsReadDescriptor): Filter {
```

Comment:
Let's move it into `handlers/reads-read.ts` now that this is only used by the handler.